source/common/config/grpc_mux_impl.cc

```diff
@@ -70,8 +71,10 @@ void GrpcMuxImpl::sendDiscoveryRequest(const std::string& type_url) {
 
 GrpcMuxWatchPtr GrpcMuxImpl::addWatch(const std::string& type_url,
                                       const std::set<std::string>& resources,
-                                      SubscriptionCallbacks& callbacks) {
-  auto watch = std::make_unique<GrpcMuxWatchImpl>(resources, callbacks, type_url, *this);
+                                      SubscriptionCallbacks& callbacks,
+                                      OpaqueResourceDecoder& resource_decoder) {
```

Comment:
As above, since this has the type-specific `Any` conversion internally, we need to supply the resource decoder on each watch when muxing across types, i.e. ADS.

Reply:
got it